File: CPP [2001-2500]/2490.circular_sentence [1].cpp
class Solution {
public:
    // T.C.=O(n), S.C.=O(1)
    bool isCircularSentence(string sentence) {
        int n = sentence.length();

        if(sentence[0] != sentence[n-1])
            return false;

        for(int i=0; i<n; i++)
            if(sentence[i] == ' ' && sentence[i-1] != sentence[i+1])
                return false;

        return true;
    }
};

File: JS/2490.circular_sentence [1].js
/**
 * @param {string} sentence
 * @return {boolean}
 */
var isCircularSentence = function(sentence) {
    if(sentence[0] !== sentence.at(-1))
        return false

    const words = sentence.split(' ')

    for(let i=1; i<words.length; i++)
        if(words[i-1].at(-1) !== words[i][0])
            return false

    return true
};